languages/js/src/Host.ts

```diff
@@ -317,16 +315,27 @@ export class Host implements Required<DataFilteringQueryParams> {
   ): Promise<boolean> {
     let instance = this.getInstance(id);
     instance = instance instanceof Promise ? await instance : instance;
-    if (!isObj(instance)) return false;
-    const mro = ancestors(instance.constructor);
-    const leftIndex = mro.indexOf(this.getClass(left));
-    const rightIndex = mro.indexOf(this.getClass(right));
-    if (leftIndex === -1) {
-      return false;
-    } else if (rightIndex === -1) {
-      return true;
-    } else {
-      return leftIndex < rightIndex;
+    try {
+      // NOTE(gj): TS is worried about looking up the `.constructor` property
+      // on an `unknown` type. `instance.constructor` will throw a `TypeError`
+      // if `instance` is `null` or `undefined`, but we catch that error down
+      // below and return `false`.
+      //
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
just want to make sure I understand ... this is a comment telling the linter not to care about a comment telling the type checker not to care about the next line?

Reply:
Can you get the same effect with
```javascript
if (instance === null || instance === undefined) throw new Error();
```

Reply:
> just want to make sure I understand ... this is a comment telling the linter not to care about a comment telling the type checker not to care about the next line?

Yeah, that's it. We have [this typescript-eslint rule](https://github.com/typescript-eslint/typescript-eslint/blob/02c6ff3c5a558f9308d7166d524156dc12e32759/packages/eslint-plugin/docs/rules/ban-ts-comment.md) enabled.

> Can you get the same effect with
> 
> ```js
> if (instance === null || instance === undefined) throw new Error();
> ```

I did something similar to this in 2725ad6. What do you think?

Reply:
Got confirmation that this is 👍